main.go

```diff
@@ -135,6 +138,7 @@ func main() {
 			Client: mgr.GetClient(),
 			Log:    ctrl.Log.WithName("controllers").WithName("KataConfig"),
 			Scheme: mgr.GetScheme(),
+			// imageGenerator: &controllers.NewImageGenerator(mgr.GetClient()), TODO: legit? add later
```

Comment:
If this is only useful for peer pods, I'd say no.

Reply:
It is, i suggested as it is much cleaner to initialize it there, Delete/Create will be still called only if peerpods is enable